src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { FC, Children, useState, useEffect } from 'react';
+import styled from 'styled-components';
+import IconButton from '../IconButton';
+import chevronRight from '../../assets/icons/chevron-right.svg';
+import chevronLeft from '../../assets/icons/chevron-left.svg';
+
+const OuterWrapper = styled.div<{ ratio: [number, number] }>`
+    position: relative;
+    overflow: hidden;
+    width: 100%;
+    padding-top: ${({ ratio }) => 100 / (ratio[0] / ratio[1])}%;
+`;
+
+const Slide = styled.div<{ active: boolean; x: string }>`
+    position: absolute;
+    top: 0;
+    width: 100%;
+    height: 100%;
+    z-index: ${({ active }) => (active ? '1' : '')};
+    transform: translateX(${({ x }) => x});
+    transition-timing-function: ease;
+    transition: transform 0.3s;
+`;
+
+const SlideButton = styled(IconButton)<{ direction: 'prev' | 'next' }>`
+    position: absolute;
+    top: calc(50% - 21px);
+    ${({ direction }) => (direction === 'next' ? 'right: 12px' : 'left: 12px')}
+    background: rgba(255,255,255, 0.6);
+    border-radius: 50%;
+    z-index: 3;
+    width: 42px;
+    height: 42px;
+`;
+
+type PropsType = {
+    'data-testid'?: string;
+    /**
+     * Let's you define the aspect ratio of the slides in the carousel, default: 16,9
+     */
+    ratio?: [number, number];
+    /**
+     * This component has 2 modes, controlled and uncontrolled.
+     * You can manage the state yourself by passing in the active slide
+     * and adding an onChange callback
+     */
+    slide?: number;
+    /**
+     * Allows you to add a callback. It'll receive the slide that should be shown **next**.
+     */
+    onChange?(newSlide: number): void;
```

Comment:
Since this callback is called **before** the next slide is animated in maybe it's better to explicitly name this callback like that. I think this is a nicer name since the exact moment when a callback is fired could be important in some cases.

```suggestion
    onBeforeChange?(newSlide: number): void;
```

Reply:
This is a bit of a tricky one, because you also want to use this callback to control the state of the current slide if you need to control the carousel:

```jsx
<Carousel 
    slide={slide}
    onChange={(newSlide) => {
        setSlide(newSlide);
    }} 
/>
```
I think in this scenario I think `onChange` is the most obvious naming (since we use that on a plethora of other components).

I definitely think animation based callbacks like onBeforeChange/onAfterChange can be useful. I can imagine a case like:

```jsx
<Carousel 
    slide={slide}
    onChange={(newSlide, oldSlide) => {
        setSlide(newSlide);
    }} 
    onBeforeChange={(newSlide, oldSlide) => setAnimating(true)}
    onAfterChange={(newSlide, oldSlide) => setAnimating(false)}
/>
<Button title={`Buy product ${products[slide].title}`} disabled={isAnimating} />
```

I suggest we both implement these, what do you think @Dreamdealer ?